packages/drivers/odsp-driver/src/odspCache.ts

```diff
@@ -124,4 +132,17 @@ export class NonPersistentCache implements INonPersistentCache {
 	>();
 
 	public readonly fileUrlCache = new PromiseCache<string, IOdspResolvedUrl>();
+
+	public readonly snapshotPrefetchResultCache = new PromiseCache<
+		string,
+		ISnapshotContentsWithEpoch
+	>();
+}
+
+export interface ISnapshotContentsWithEpoch extends ISnapshotContents {
+	fluidEpoch: string;
+}
+
+export function snapshotPrefetchCacheKeyFromEntry(entry: ICacheEntry): string {
+	return `${entry.file.docId}_${entry.type}_${entry.key}`;
```

Comment:
This API is a bit weird, because it assumes that it will be called for entry.type ==- "snapshot", but it does not assert that.
And it's not very clear from API alone on what it does - one needs to look at usage to realize that it provides a mapping between persistent cache (snapshot) key to a key in prefetchCache.

I think we can be served just fine by more generic function (name) doing same thing, like existing getKeyForCacheEntry(), It would be also great to make LocalPersistentCache.keyFromEntry(), OdspPersistentCache.keyFromEntry(), OdspSampleCache.keyFromEntry() to not reimplement that same logic but rather call directly into getKeyForCacheEntry()

Reply:
done